### PR TITLE
fix(chat): Fix error if no recent models set (Issue #2653)

### PR DESCRIPTION
**Pull request by denys-kolomiitsev**
**Description:**

Fix error if no recent models set

Issues:

- Issue #2653 

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2729.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/store/models/models.epics.ts [hunk 1]
```diff
@@ -236,14 +236,21 @@ const getInstalledModelIdsFailEpic: AppEpic = (action$, state$) =>
       const recentModelIds = ModelsSelectors.selectRecentModelsIds(
         state$.value,
       );
+      const availableModels = ModelsSelectors.selectModels(state$.value);
+      const firstAvailableModelId = availableModels[0].id;
```
Comment (IlyaBondar) [inline apps/chat/src/store/models/models.epics.ts]: `availableModels[0]` can be undefined

Comment (denys-kolomiitsev) [reply apps/chat/src/store/models/models.epics.ts]: Thank you for the comment.
Added fallback.

File: apps/chat/src/store/models/models.epics.ts [hunk 2]
```diff
@@ -236,14 +236,23 @@ const getInstalledModelIdsFailEpic: AppEpic = (action$, state$) =>
       const recentModelIds = ModelsSelectors.selectRecentModelsIds(
         state$.value,
       );
+      const availableModels = ModelsSelectors.selectModels(state$.value);
+      const firstAvailableModelId = availableModels?.[0].id;
```
Comment (IlyaBondar) [inline apps/chat/src/store/models/models.epics.ts]: availableModels is empty array by default, [0] can be undefined
```suggestion
      const firstAvailableModelId = availableModels?.[0]?.reference;
```

Comment (Alexander-Kezik) [reply apps/chat/src/store/models/models.epics.ts]: Also it’s better to take a reference instead of id

Comment (denys-kolomiitsev) [reply apps/chat/src/store/models/models.epics.ts]: Suggestion applied.
Thank you!

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2729.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/store/models/models.epics.ts [hunk 3]
```diff
@@ -236,14 +236,23 @@ const getInstalledModelIdsFailEpic: AppEpic = (action$, state$) =>
...
 
+      const installCandidates = [...myAppIds, ...modelsToInstall];
+      const agentsToInstall = installCandidates.length
+        ? installCandidates
+        : fallbackModels;
+
       return of(
         ModelsActions.addInstalledModels({
-          references: [...myAppIds, ...modelsToInstall],
+          references: [...agentsToInstall],
```
Comment (IlyaBondar) [inline apps/chat/src/store/models/models.epics.ts]:
```suggestion
          references: agentsToInstall,
```

Comment (denys-kolomiitsev) [reply apps/chat/src/store/models/models.epics.ts]: Fixed

**Review by IlyaBondar — APPROVED** [review]

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2729.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2729.nightly-test.deltixhub.io
>E2E tests status: success

**Pull request merged**